web/src/components/profile/ProfileNewJerseyAddress.tsx

```diff
@@ -52,87 +56,97 @@ export const ProfileNewJerseyAddress = (): ReactElement => {
+      ? true
+      : false;
+  };
+
   return (
     <>
-      <div className="margin-y-4">
-        <div id={`question-addressLine1`} className="text-field-width-default add-spacing-on-ele-scroll">
-          <AddressTextField
-            label={Config.formation.fields.addressLine1.label}
-            fieldName="addressLine1"
-            validationText={getFieldErrorLabel("addressLine1")}
-            className={"margin-bottom-2"}
-            errorBarType="ALWAYS"
-            onValidation={onValidation}
-          />
-        </div>
-        <div id={`question-addressLine2`} className="text-field-width-default add-spacing-on-ele-scroll">
-          <AddressTextField
-            label={Config.formation.fields.addressLine2.label}
-            secondaryLabel={Config.formation.general.optionalLabel}
-            errorBarType="ALWAYS"
-            fieldName="addressLine2"
-            validationText={getFieldErrorLabel("addressLine2")}
-            className="margin-bottom-2"
-            onValidation={onValidation}
-          />
-        </div>
-        <div className="text-field-width-default">
-          <WithErrorBar
-            hasError={doSomeFieldsHaveError(["addressState", "addressZipCode", "addressMunicipality"])}
-            type="DESKTOP-ONLY"
-          >
-            <div className="grid-row tablet:grid-gap-2">
-              <div className="grid-col-12 tablet:grid-col-6">
-                <WithErrorBar hasError={doesFieldHaveError("addressMunicipality")} type="MOBILE-ONLY">
-                  <span className="text-bold">{Config.formation.fields.addressCity.label}</span>
-                  <AddressMunicipalityDropdown onValidation={onValidation} />
-                </WithErrorBar>
-              </div>
-              <div className="grid-col-12 tablet:grid-col-6">
-                <WithErrorBar
-                  hasError={doSomeFieldsHaveError(["addressState", "addressZipCode"])}
-                  type="MOBILE-ONLY"
-                >
-                  <div className="grid-row grid-gap tablet:grid-gap-2">
-                    <div className="grid-col-6">
-                      <strong>
-                        <ModifiedContent>{Config.formation.fields.addressState.label}</ModifiedContent>
-                      </strong>
-                      <div
-                        id={`question-addressState`}
-                        className="text-field-width-default add-spacing-on-ele-scroll"
-                      >
-                        <StateDropdown
-                          fieldName="addressState"
-                          value={"New Jersey"}
-                          validationText={Config.formation.fields.addressState.error}
-                          disabled={true}
-                          onSelect={(): void => {}}
-                        />
+      {areAddressFieldsDisabled() ? (
+        <ProfileAddressLockedFields />
+      ) : (
```

Comment:
Pebble - everything that is rendered below should be it's own component

Reply:
We can refactor this on a Friday.

Reply:
Pivotal Tracker refactor ticket: [Refactor ProfileNewJerseyAddress Component to Separate Address Fields into a Different Component](https://www.pivotaltracker.com/n/projects/2573455/stories/188108775)

Reply:
I don't know that I agree that we should create yet another component here. The purpose of this component is specifically to be that it is responsible for the address fields.

I would rather see us unify all instances where we handle and address throughout the app instead of continuing to make smaller pieces.